docs: document critical frontend pattern migrations in CLAUDE.md

mafredri opened the pull request:
## Summary

This PR documents critical frontend pattern migrations based on **systematic analysis of 200+ git commits** that modified `site/` files. These patterns have been causing repeated feedback in PR reviews and need to be clearly documented for AI and human developers.

## Analysis Method

Used `git log` to identify actual pattern migrations:
- Analyzed 200+ commits touching `site/` directory
- Examined actual code diffs for migration patterns
- Reviewed PR discussions via GitHub API
- Identified systematic pattern changes across multiple PRs

## Patterns Documented

### 1. **Emotion → Tailwind CSS**
- **Policy**: "No new emotion styles, full stop"
- Documents the strict migration away from `@emotion/react` to Tailwind CSS
- Includes concrete examples of common conversions
- Emphasizes refactoring old code when touching it
- **PRs analyzed**: #20530, #20479, #20204, #20946

### 2. **MUI Components → Custom/Radix/shadcn**
- Documents replacement of Material-UI components with custom equivalents
- Specific guidance for Tooltip migration (100ms delay, Radix-based)
- Instructions for systematic replacement of Tables, Buttons
- **PRs analyzed**: #20849, #20869, #20261, #20201, #20200, #19968, #20938

### 3. **MUI Icons → lucide-react** (NEW)
- All icons must use lucide-react, never MUI icons
- Specific icon mappings documented:
  - `BusinessIcon` → `Building2Icon`
  - `GroupOutlinedIcon` → `UsersIcon`
  - `PublicOutlinedIcon` → `GlobeIcon`
  - `PersonIcon` → `UserIcon`
- **PRs analyzed**: #20193 (14 files changed), #20318

### 4. **Radix Component Prop Naming** (NEW)
- Documents prop name changes when migrating MUI → Radix
- `placement` → `side`
- `PopperProps` removed (internal implementation detail)
- Composition pattern with Provider/Root/Trigger/Content
- **PR analyzed**: #20261

### 5. **Testing: spyOn → queries parameter**
- Preferred pattern for GET endpoint mocks in Storybook stories
- Detailed before/after examples
- Clarifies when spyOn is still appropriate
- **PR analyzed**: #20948

### 6. **State Storage: localStorage → user_configs**
- Documents that user preferences should be stored in backend
- Explains why localStorage is browser-specific, not user-specific
- Provides `theme_preference` as reference implementation
- **PR analyzed**: #20900

### 7. **Chromatic Testing Best Practices**
- Prefer visual validation over programmatic assertions
- Avoid redundant assertions that duplicate snapshot validation
- Reduce test flakiness
- **PR analyzed**: #20808

### 8. **cn() Utility for Conditional Classes** (NEW)
- Standard pattern for conditional className merging
- Combines `clsx` and `tailwind-merge`
- Used throughout shadcn components

## Changes Made

- Updated `site/CLAUDE.md` with comprehensive pattern documentation
- Added prominent "Critical Pattern Migrations" section (now 5 critical patterns)
- Included code examples for each pattern
- Added new sections for icon migration and Radix conventions
- Emphasized "leave the campsite better than you found it" philosophy

## Git Statistics

- **Commits analyzed**: 200+ touching `site/` directory
- **Refactor/migration PRs found**: 15+
- **Pattern-related PRs**: 10+
- **Consistent migration effort**: October-November 2025

## Key PRs Referenced

#20948, #20946, #20938, #20933, #20905, #20900, #20869, #20849, #20808, #20530, #20479, #20318, #20261, #20204, #20201, #20200, #20193

---

🤖 This change was written by Claude Sonnet 4.5 Thinking using mux and reviewed by a human 🏂

mafredri committed:
docs(site): document critical frontend pattern migrations

Add comprehensive documentation for active frontend pattern migrations
discovered through analysis of 200+ git commits touching site/ files:

Core Migrations:
- Emotion → Tailwind CSS (strict 'no new emotion' policy)
- MUI Components → Custom/Radix/shadcn (Tooltips, Tables, Buttons)
- MUI Icons → lucide-react with specific icon mappings
- spyOn → queries parameter for GET endpoint mocks in Storybook
- localStorage → user_configs table for user preferences

New Documentation:
- Icon migration mappings (BusinessIcon→Building2Icon, etc.)
- Radix component prop naming conventions (placement→side)
- cn() utility usage for conditional className merging
- Chromatic testing best practices (prefer snapshots over assertions)

Includes concrete before/after examples and migration patterns to guide
developers away from deprecated approaches toward current best practices.

Analysis based on PRs: #20948, #20946, #20938, #20905, #20900, #20869,
#20849, #20808, #20530, #20479, #20261, #20201, #20200, #20193, #20318

---

🤖 This change was written by Claude Sonnet 4.5 Thinking using mux and reviewed by a human 🏂

mafredri committed:
fix(docs): add blank lines around lists for markdown linter

Fix markdown linting errors MD032 and MD031:
- Add blank lines around lists (MD032)
- Add blank lines around fenced code blocks (MD031)

Ensures site/CLAUDE.md passes markdownlint-cli2 checks.

aslilac reviewed (commented):
seems to mostly make sense

one small worry: having the "don't"s in here puts them in the context window, and I worry that might some models might pick up on that and just ignore the "don't". I wonder if it might be better to _only_ put the new equivalent? totally guessing tho

@jaaydenh has more experience with this kind of tuning so I'd love to here from him

mafredri committed:
refactor(docs): remove deprecated pattern examples to prevent AI misuse

Remove code examples showing deprecated patterns (OLD/DO NOT USE blocks)
that could mislead AI agents into copying the wrong patterns.

Changes:
- Icon section: Only show lucide-react imports, not MUI icons
- Radix section: Only show correct Radix pattern, not MUI pattern
- Emotion section: Only show Tailwind classes, not emotion imports
- Testing section: Only show queries parameter, not spyOn pattern

This prevents AI agents from pattern-matching and copying deprecated
code while maintaining clear guidance on what to replace and how.

mafredri committed:
refactor(docs): reduce wordiness in pattern documentation

Condense verbose sections to be more scannable and direct:
- Icon section: Single-line import, inline replacement mappings
- Radix section: Consolidated prop changes into one sentence
- Emotion section: Reduced from 5 to 4 bullets, combined redundant points
- Chromatic section: Reduced from 5 to 3 bullets
- localStorage section: Reduced from 8 to 4 lines

Overall ~30% reduction in text while maintaining clarity. AI agents
scan for patterns and use code examples - keeping prose minimal and
directive improves efficiency.

mafredri committed:
refactor(docs): remove duplicate content from site/CLAUDE.md

Remove redundant sections and repeated information:

- Remove entire "Migration" section (lines 192-207) that duplicated
  Components, Styling, and Tailwind sections
- Remove duplicate "leave campsite better" mention (already in line 13)
- Remove duplicate "Never import from @emotion/react" (already stated)

Reduces documentation from 250 to 232 lines while maintaining all
unique information. Each concept is now documented once in its most
appropriate section.

mafredri committed:
style(docs): remove emoji to match original doc style

Remove 🚨 emoji from "Critical Pattern Migrations" section header to
maintain consistency with the original document style which uses no
emojis, only text emphasis and formatting.

All other style elements match the original:
- Section headers with ## and ###
- Bold emphasis for important terms
- Simple bullet lists with -
- Direct, imperative tone
- "(MUST FOLLOW)" matches original "(USE FIRST)" pattern

jaaydenh commented on site/CLAUDE.md:
```diff
@@ -26,25 +38,84 @@ When investigating or editing TypeScript/React code, always use the TypeScript l
 
+### Icon Migration: MUI Icons → lucide-react
+
+Never import from `@mui/icons-material`. Use `lucide-react` instead.
```
@mui/icons-material is no longer used in the codebase, I think its highly unlikely the AI will suggest using a MUI icon anymore without any usages in the codebase.

jaaydenh reviewed (approved):
I think the only change needed is removing the section about MUI Icons
